commit cd105773505eaa61614502953b1f738298426487
Author: Baptiste Jonglez <git@bitsofnetworks.org>
Date:   Sat Jul 15 19:48:09 2017 +0200

backport to LEDE-17.01

    x86: Fix xen serial console by removing conflicting PATA driver
    
    The Xen serial console has been broken since the xen_domu subtarget
    was merged in the generic x86 subtarget (commits 1d6879ee and 371b382a).
    
    The reason for the broken serial console seems to be an IRQ conflict
    between the serial console driver and the PATA_LEGACY driver:
    
    [    1.330125] genirq: Flags mismatch irq 8. 00000000 (hvc_console) vs. 00000000 (platform[pata_legacy.4])
    [    1.330134] hvc_open: request_irq failed with rc -16.
    [    1.330148] Warning: unable to open an initial console.
    
    Just drop the PATA_LEGACY driver from the x86/generic and x86_64
    subtargets, since this driver is marked experimental and only supports
    very old ISA devices anyway.  It is still included in the x86/legacy
    subtarget where it rightfully belongs.
    
    Fixes: FS#787
    
    Signed-off-by: Baptiste Jonglez <git@bitsofnetworks.org>

diff --git a/target/linux/x86/64/config-default b/target/linux/x86/64/config-default
index 35336e4..1a3f915 100644
--- a/target/linux/x86/64/config-default
+++ b/target/linux/x86/64/config-default
@@ -242,7 +242,6 @@ CONFIG_PARAVIRT_CLOCK=y
 CONFIG_PARAVIRT_SPINLOCKS=y
 CONFIG_PATA_AMD=y
 CONFIG_PATA_ATIIXP=y
-CONFIG_PATA_LEGACY=y
 CONFIG_PATA_MPIIX=y
 CONFIG_PATA_OLDPIIX=y
 CONFIG_PATA_PLATFORM=y
diff --git a/target/linux/x86/generic/config-default b/target/linux/x86/generic/config-default
index 310c37c..fa430cf 100644
--- a/target/linux/x86/generic/config-default
+++ b/target/linux/x86/generic/config-default
@@ -247,7 +247,6 @@ CONFIG_PARAVIRT_CLOCK=y
 CONFIG_PARAVIRT_SPINLOCKS=y
 CONFIG_PATA_AMD=y
 CONFIG_PATA_ATIIXP=y
-CONFIG_PATA_LEGACY=y
 CONFIG_PATA_MPIIX=y
 CONFIG_PATA_OLDPIIX=y
 CONFIG_PATA_PLATFORM=y
